pkg/proc/breakpoints.go

```diff
@@ -384,6 +390,60 @@ func (t *Target) SetBreakpoint(addr uint64, kind BreakpointKind, cond ast.Expr)
 	return t.setBreakpointInternal(addr, kind, 0, cond)
 }
 
+func (t *Target) SetTracepoint(fnName string) error {
+	if !t.proc.SupportsBPF() {
+		return errors.New("eBPF is not supported")
+	}
+	var args []ebpf.UProbeArgMap
+	fn, ok := t.BinInfo().LookupFunc[fnName]
+	if !ok {
+		return fmt.Errorf("could not find function %s", fnName)
+	}
+	dwarfTree, err := fn.cu.image.getDwarfTree(fn.offset)
+	if err != nil {
+		return err
+	}
+	variablesFlags := reader.VariablesOnlyVisible
+	if t.BinInfo().Producer() != "" && goversion.ProducerAfterOrEqual(t.BinInfo().Producer(), 1, 15) {
+		variablesFlags |= reader.VariablesTrustDeclLine
+	}
+	_, l, _ := t.BinInfo().PCToLine(fn.Entry)
+
+	regs, err := t.CurrentThread().Registers()
+	if err != nil {
+		return err
+	}
+	dregs := t.BinInfo().Arch.RegistersToDwarfRegisters(fn.cu.image.StaticBase, regs)
+	pcreg := dregs.Reg(t.BinInfo().Arch.PCRegNum)
+	pcreg.Uint64Val = fn.Entry
+
+	varEntries := reader.Variables(dwarfTree, fn.Entry, l, variablesFlags)
+	for _, entry := range varEntries {
+		isret, _ := entry.Val(dwarf.AttrVarParam).(bool)
+		if isret {
+			continue
+		}
+		_, dt, err := readVarEntry(entry.Tree, fn.cu.image)
+		if err != nil {
+			return err
+		}
+		offset, pieces, _, err := t.BinInfo().Location(entry, dwarf.AttrLocation, fn.Entry, *dregs)
```

Comment:
I think you should be able to pass `op.DwarfRegisters{}` here. That's better because we are not using the registers of the current thread that have nothing to do with what we are doing anyway.

Reply:
Done.